providers/Flipt/src/FliptProvider.php

```diff
@@ -0,0 +1,137 @@
+        $this->client = ( is_string( $hostOrClient ) ) ? new FliptClient( $hostOrClient, $apiToken, $namespace ) : $hostOrClient;
+        $this->cache = new Cache( $cache );
+    }
+
+
+    public function resolveBooleanValue(string $flagKey, bool $defaultValue, ?EvaluationContext $context = null): ResolutionDetails
+    {
+        return $this->resolveValue($flagKey, FlagValueType::BOOLEAN, $defaultValue, $context);
+    }
+
+    public function resolveStringValue(string $flagKey, string $defaultValue, ?EvaluationContext $context = null): ResolutionDetails
+    {
+        return $this->resolveValue($flagKey, FlagValueType::STRING, $defaultValue, $context);
+    }
+
+    public function resolveIntegerValue(string $flagKey, int $defaultValue, ?EvaluationContext $context = null): ResolutionDetails
+    {
+        return $this->resolveValue($flagKey, FlagValueType::INTEGER, $defaultValue, $context);
+    }
+
+    public function resolveFloatValue(string $flagKey, float $defaultValue, ?EvaluationContext $context = null): ResolutionDetails
+    {
+        return $this->resolveValue($flagKey, FlagValueType::FLOAT, $defaultValue, $context);
+    }
+
+    /**
+     * @param mixed[] $defaultValue
+     */
+    public function resolveObjectValue(string $flagKey, array $defaultValue, ?EvaluationContext $context = null): ResolutionDetails
+    {
+        return $this->resolveValue($flagKey, FlagValueType::OBJECT, $defaultValue, $context);
+    }
+
+
+    /**
+     * Clears the cache of all requests
+     */
+    public function clearCache() {
+        $this->cache->clear();
+    }
+
+    /**
+     * @param bool|string|int|float|mixed[] $defaultValue
+     */
+    private function resolveValue(string $flagKey, string $flagType, mixed $defaultValue, ?EvaluationContext $context = null): ResolutionDetails
+    {
+
+        // check null context
+        if( empty( $context ) ) {
+            $attributes = [];
+            $id = null;
+        } else {
+            $attributes = $context->getAttributes()->toArray();
+            $id = $context->getTargetingKey();
+        }
+
+
+        // check if cache has already result
+        $cacheKey = $this->cache->key( [ 'flag' => $flagKey, 'type' => $flagType, 'default' => $defaultValue, 'context' => $attributes, 'key' => $id ] );
+        $cached = $this->cache->get( $cacheKey );
+        if( isset( $cached ) ) return $cached;
+
+
+        // booleans need a dedicated function
+        if( $flagType == FlagValueType::BOOLEAN ) {
+            $result = $this->client->boolean( $flagKey, $attributes, $id );
+        } else {
+            $result = $this->client->variant( $flagKey, $attributes, $id );
+        }
+
+        
+        // there is a match
+        // not sure yet as the variant result has a getMatch() but not the boolean result.
+        if( $result->getReason() == 'MATCH_EVALUATION_REASON' || $result->getReason() == "DEFAULT_EVALUATION_REASON" ) {
```

Comment:
Could we set these strings up as consts, if they aren't available from the Flipt SDK directly?